lib/dash/dash_parser.js

```diff
@@ -556,23 +555,22 @@ shaka.dash.DashParser = class {
     const periods = [];
     let prevEnd = 0;
     const periodNodes = XmlUtils.findChildren(mpd, 'Period');
-    // This uses a for-loop rather than a for-of loop because this needs to look
-    // ahead to the next element.
-    const enumerate = (it) => shaka.util.Iterables.enumerate(it);
-    for (const {i, item: elem, next} of enumerate(periodNodes)) {
+    for (let i = 0; i < periodNodes.length; i++) {
+      const elem = periodNodes[i];
       const start = /** @type {number} */ (
         XmlUtils.parseAttr(elem, 'start', XmlUtils.parseDuration, prevEnd));
       const periodId = elem.id;
       const givenDuration =
           XmlUtils.parseAttr(elem, 'duration', XmlUtils.parseDuration);
 
       let periodDuration = null;
-      if (next) {
+      if (i + 1 < periodNodes.length) {
```

Comment:
Repetition of `i + 1 < periodNodes.length` could probably be avoided by having a `const hasNext = i + 1 < periodNodes.length;`

Reply:
TBH only the last iteration won't have a next so I'd suggest just doing
```
const elem = periodNodes[i];
const next = periodNodes[i + 1];
```
and leaving the code as it was

Reply:
Done!